class-auth.php

```diff
@@ -178,7 +178,31 @@ public function get_token( WP_REST_Request $request ) {
 			);
 		}
 
-		if ( isset( $_COOKIE['refresh_token'] ) ) {
+		if ( isset( $username ) && ! isset( $password )) {
+			$user = new WP_Error(
+				'jwt_auth_no_password',
+				__( 'Password is required', 'jwt-auth' ),
+				array(
+					'status' => 400,
+				)
+			);
+		}
+
+		if ( isset( $password ) && ! isset( $username )) {
+			$user = new WP_Error(
+				'jwt_auth_no_username',
+				__( 'Username is required', 'jwt-auth' ),
+				array(
+					'status' => 400,
+				)
+			);
+		}
+
+		if ( isset( $password ) && isset( $username )) {
```

Comment:
Forms will send an empty string if fields are left blank, so let's keep the new "one of both missing check" this simpler.

```suggestion
		if ( ( isset( $username ) && ! isset( $password ) )
		     ( ! isset( $username ) && isset( $password ) ) ) {
			$user = new WP_Error(
				'jwt_auth_missing_credentials',
				__( 'Username and password are required', 'jwt-auth' ),
				array(
					'status' => 400,
				)
			);
		}

		if ( isset( $username ) && isset( $password ) ) {
```

The new error code would also have to be documented in the readme.

Reply:
Accepted this, but also added a check for no refresh token here as well, otherwise there's nothing to catch cases where there's no username and password OR refresh token.